src/migrateCalculatedMeasuresToActivePivot/migrateCalculatedMeasuresInWidgets.test.ts

```diff
@@ -0,0 +1,112 @@
+import { migrateCalculatedMeasuresInWidgets } from "./migrateCalculatedMeasuresInWidgets";
+import { uiWidgetsFolder } from "../__test_resources__/aui5.0LegacyTestResources/uiWidgetsFolder";
+import { dataModelsForTests } from "@activeviam/data-model";
+
+const dataModel = dataModelsForTests.sandbox;
+
+// "pvSum ^ 2" is from cube "EquityDerivativesCubeDist", all others are from "EquityDerivativesCube".
+const namesOfCalculatedMeasurestoMigrate = [
+  "Distinct count city",
+  "Test calculated measure",
+  "EXP pnl.Forex",
+  "Log pv.SUM",
+  "pvSum ^ 2",
+];
+
+describe("migrateCalculatedMeasuresInWidgets", () => {
+  const { cubeNames, migratedWidgetsRecord } =
+    migrateCalculatedMeasuresInWidgets(
+      uiWidgetsFolder,
+      dataModel,
+      namesOfCalculatedMeasurestoMigrate,
+    );
+
+  it("returns a `cubeNames` object containing the names of all calculated measures used in a `ui/widgets` folder with their corresponding cube name", () => {
+    expect(cubeNames).toStrictEqual({
+      "Distinct count city": "EquityDerivativesCube",
+      "EXP pnl.Forex": "EquityDerivativesCube",
+      "Log pv.SUM": "EquityDerivativesCube",
+      "Test calculated measure": "EquityDerivativesCube",
+      "pvSum ^ 2": "EquityDerivativesCubeDist",
+    });
+  });
+
+  it("returns the widget record unchanged if a widget at the root contains no calculated measures", () => {
```

Comment:
Nitpick but what is returned is the `widgets` folder as a whole, not one individual widget record:
```suggestion
  it("does not change a widget record if there is no calculated measures in its MDX query", () => {
```